# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1346510805 14400
#      Sat Sep 01 10:46:45 2012 -0400
# Node ID dd794aa84232439db94a48fe22038cc4ba078a7b
# Parent  648dabbb4c6b0a02aef79b530362ca0afe7f622e
additional changes for merger of libcruft with liboctave

* libgui/src/Makefile.am (AM_CPPFLAGS): Include
-I../../liboctave/numeric, -I../../liboctave/operators,
--I$(srcdir)/../../liboctave/array,
--I$(srcdir)/../../liboctave/numeric,
--I$(srcdir)/../../liboctave/operators,
--I$(srcdir)/../../liboctave/system, and
-I$(srcdir)/../../liboctave/util in the list.
* libinterp/link-deps.mk (LIBOCTINTERP_LINK_DEPS): Delete
$(GNULIB_LINK_DEPS) from the list.
* liboctave/Makefile.am (liboctae_la_LIBADD): Include
cruft/ranlib/libranlib.la and ../libgnu/libgnu.la in the list.
* liboctave/cruft/link-deps.mk: Delete.
* liboctave/cruft/Makefile.am (noinst_LTLIBRARIES): Rename from
octlib_LTLIBRARIES.  Don't include link-deps.mk.
(libcruft_la_LIBADD, libcruft_current, libcruft_revision,
libcruft_age, libcruft_la_LDFLAGS): Delete.
Build libranlib.la in ranlib subdirectory.
* module.mk (ranlib_libranlib_la_SOURCES): Rename from libranlib_la_SOURCES.
* liboctave/link-deps.mk: Don't include
$(top_srcdir)/liboctave/cruft/link-deps.mk.
(GNULIB_LINK_DEPS): Move definition here from
liboctave/cruft/link-deps.mk.
(LIBOCTAVE_LINK_DEPS): Include $(FLIBS) in the list.
Don't add $(LIBCRUFT_LINK_DEPS) to LIBOCTAVE_LINK_DEPS.
Don't add $(LIBCRUFT_LINK_OPTS) to LIBOCTAVE_LINK_OPTS.
* liboctave/operators/module.mk: Use correct names for mk-ops.awk
script and mx-ops, vx-ops, and sparse-mx-ops files.
* src/Makefile.am (AM_CPPFLAGS): Include
-I$(srcdir)/../liboctave/array, -I$(srcdir)/../liboctave/numeric, and
-I$(srcdir)/../liboctave/util in the list.
Include -I$(srcdir)/../liboctave/cruft/misc in the list, not
-I$(srcdir)/../libcruft/misc.

diff --git a/libgui/src/Makefile.am b/libgui/src/Makefile.am
--- a/libgui/src/Makefile.am
+++ b/libgui/src/Makefile.am
@@ -98,19 +98,26 @@ liboctgui_la_CPPFLAGS = \
   -I$(QT_INCDIR) \
   -I$(QT_INCDIR)/QtCore \
   -I$(QT_INCDIR)/QtGui \
   -I$(QT_INCDIR)/QtNetwork \
   -I$(srcdir)/../qterminal/libqterminal \
   -I$(srcdir)/m-editor \
   -I$(srcdir)/octave-adapter \
   -I$(srcdir)/qtinfo \
+  -I../../liboctave \
+  -I../../liboctave/numeric \
+  -I../../liboctave/operators \
+  -I$(srcdir)/../../liboctave \
+  -I$(srcdir)/../../liboctave/array \
   -I$(srcdir)/../../liboctave/cruft/misc \
-  -I../../liboctave \
-  -I$(srcdir)/../../liboctave \
+  -I$(srcdir)/../../liboctave/numeric \
+  -I$(srcdir)/../../liboctave/operators \
+  -I$(srcdir)/../../liboctave/system \
+  -I$(srcdir)/../../liboctave/util \
   -I../../libinterp \
   -I../../libinterp/interp-core \
   -I../../libinterp/interpfcn \
   -I../../libinterp/parse-tree \
   -I$(srcdir)/../../libinterp \
   -I$(srcdir)/../../libinterp/interp-core \
   -I$(srcdir)/../../libinterp/interpfcn \
   -I$(srcdir)/../../libinterp/octave-value \
diff --git a/libinterp/link-deps.mk b/libinterp/link-deps.mk
--- a/libinterp/link-deps.mk
+++ b/libinterp/link-deps.mk
@@ -2,17 +2,16 @@ include $(top_srcdir)/liboctave/link-dep
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
   LIBOCTINTERP_LINK_DEPS =
 else
   LIBOCTINTERP_LINK_DEPS = $(DLDFCN_LIBS)
 endif
 
 LIBOCTINTERP_LINK_DEPS += \
-	$(GNULIB_LINK_DEPS) \
   $(GRAPHICS_LIBS) \
   $(FT2_LIBS) \
   $(HDF5_LIBS) \
   $(Z_LIBS) \
   $(FFTW_XLIBS) \
   $(REGEX_LIBS) \
   $(OPENGL_LIBS) \
   $(X11_LIBS) \
diff --git a/liboctave/Makefile.am b/liboctave/Makefile.am
--- a/liboctave/Makefile.am
+++ b/liboctave/Makefile.am
@@ -23,18 +23,18 @@ include $(top_srcdir)/build-aux/common.m
 AUTOMAKE_OPTIONS = subdir-objects
 
 ## Run cruft dir with stand-alone Makefile.
 ## Eventually this will use module.mk syntax.
 SUBDIRS = cruft
 
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
+  -I$(srcdir)/array \
   -I$(srcdir)/cruft/misc \
-  -I$(srcdir)/array \
   -Inumeric -I$(srcdir)/numeric \
   -Ioperators -I$(srcdir)/operators \
   -I$(srcdir)/system \
   -I$(srcdir)/util \
   -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
 octlib_LTLIBRARIES = liboctave.la
 
@@ -89,21 +89,23 @@ liboctave_la_CPPFLAGS = \
   $(FFTW_XCPPFLAGS) \
   $(ARPACK_CPPFLAGS) \
   $(AM_CPPFLAGS)
 
 include link-deps.mk
 
 liboctave_la_LIBADD = \
   array/libarray.la \
-	cruft/libcruft.la \
-	numeric/libnumeric.la \
-	operators/liboperators.la \
-	system/libsystem.la \
-	util/libutil.la \
+  cruft/libcruft.la \
+  cruft/ranlib/libranlib.la \
+  numeric/libnumeric.la \
+  operators/liboperators.la \
+  system/libsystem.la \
+  util/libutil.la \
+  ../libgnu/libgnu.la \
   $(LIBOCTAVE_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 liboctave_current = 1
 liboctave_revision = 1
 liboctave_age = 0
 
 liboctave_version_info = $(liboctave_current):$(liboctave_revision):$(liboctave_age)
diff --git a/liboctave/cruft/Makefile.am b/liboctave/cruft/Makefile.am
--- a/liboctave/cruft/Makefile.am
+++ b/liboctave/cruft/Makefile.am
@@ -21,56 +21,35 @@
 include $(top_srcdir)/build-aux/common.mk
 
 AUTOMAKE_OPTIONS = subdir-objects
 
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
   -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
-octlib_LTLIBRARIES = libcruft.la
+noinst_LTLIBRARIES = libcruft.la
 
 libcruft_la_SOURCES =
 
 nodist_libcruft_la_SOURCES =
 
 libcruft_la_FFLAGS = $(F77_INTEGER_8_FLAG)
 
 libcruft_la_CPPFLAGS = \
   @CRUFT_DLL_DEFS@ \
   $(AM_CPPFLAGS)
 
-include link-deps.mk
-
-libcruft_la_LIBADD = \
-  libranlib.la \
-  $(top_builddir)/libgnu/libgnu.la \
-  $(LIBCRUFT_LINK_DEPS)
-
-# Increment these as needed and according to the rules in the libtool manual:
-libcruft_current = 1
-libcruft_revision = 0
-libcruft_age = 0
-
-libcruft_version_info = $(libcruft_current):$(libcruft_revision):$(libcruft_age)
-
-libcruft_la_LDFLAGS = \
-  -version-info $(libcruft_version_info) \
-  $(NO_UNDEFINED_LDFLAG) \
-  @XTRA_CRUFT_SH_LDFLAGS@ \
-  -bindir $(bindir) \
-  $(LIBCRUFT_LINK_OPTS)
-
 libcruft_la_DEPENDENCIES = cruft.def
 
-noinst_LTLIBRARIES = libranlib.la
+noinst_LTLIBRARIES += ranlib/libranlib.la
 
-libranlib_la_SOURCES =
+ranlib_libranlib_la_SOURCES =
 
-libranlib_la_DEPENDENCIES = ranlib.def
+ranlib_libranlib_la_DEPENDENCIES = ranlib.def
 
 octinclude_HEADERS =
 
 EXTRA_DIST =
 
 include amos/module.mk
 include blas-xtra/module.mk
 include daspk/module.mk
diff --git a/liboctave/cruft/link-deps.mk b/liboctave/cruft/link-deps.mk
deleted file mode 100644
--- a/liboctave/cruft/link-deps.mk
+++ /dev/null
@@ -1,23 +0,0 @@
-GNULIB_LINK_DEPS = \
-  $(COPYSIGNF_LIBM)\
-  $(COPYSIGN_LIBM)\
-  $(FLOOR_LIBM)\
-  $(GETHOSTNAME_LIB)\
-  $(LIBSOCKET)\
-  $(LIB_NANOSLEEP)\
-  $(LIB_SELECT)\
-  $(LTLIBINTL)\
-  $(ROUNDF_LIBM)\
-  $(ROUND_LIBM)\
-  $(TRUNCF_LIBM)\
-  $(TRUNC_LIBM)
-
-LIBCRUFT_LINK_DEPS = \
-	$(GNULIB_LINK_DEPS) \
-  $(LAPACK_LIBS) \
-  $(BLAS_LIBS) \
-  $(FLIBS) \
-  $(LIBS)
-
-LIBCRUFT_LINK_OPTS =
-
diff --git a/liboctave/cruft/ranlib/module.mk b/liboctave/cruft/ranlib/module.mk
--- a/liboctave/cruft/ranlib/module.mk
+++ b/liboctave/cruft/ranlib/module.mk
@@ -4,17 +4,17 @@ EXTRA_DIST += \
   ranlib/HOWTOGET \
   ranlib/README \
   ranlib/randlib.chs \
   ranlib/randlib.fdoc \
   ranlib/tstbot.for \
   ranlib/tstgmn.for \
   ranlib/tstmid.for
 
-libranlib_la_SOURCES += \
+ranlib_libranlib_la_SOURCES += \
   ranlib/advnst.f \
   ranlib/genbet.f \
   ranlib/genchi.f \
   ranlib/genexp.f \
   ranlib/genf.f \
   ranlib/gengam.f \
   ranlib/genmn.f \
   ranlib/genmul.f \
diff --git a/liboctave/link-deps.mk b/liboctave/link-deps.mk
--- a/liboctave/link-deps.mk
+++ b/liboctave/link-deps.mk
@@ -1,28 +1,36 @@
-include $(top_srcdir)/liboctave/cruft/link-deps.mk
+GNULIB_LINK_DEPS = \
+  $(COPYSIGNF_LIBM)\
+  $(COPYSIGN_LIBM)\
+  $(FLOOR_LIBM)\
+  $(GETHOSTNAME_LIB)\
+  $(LIBSOCKET)\
+  $(LIB_NANOSLEEP)\
+  $(LIB_SELECT)\
+  $(LTLIBINTL)\
+  $(ROUNDF_LIBM)\
+  $(ROUND_LIBM)\
+  $(TRUNCF_LIBM)\
+  $(TRUNC_LIBM)
 
 LIBOCTAVE_LINK_DEPS = \
-	$(GNULIB_LINK_DEPS) \
+  $(GNULIB_LINK_DEPS) \
   $(SPARSE_XLIBS) \
   $(ARPACK_LIBS) \
   $(QRUPDATE_LIBS) \
   $(FFTW_XLIBS) \
   $(LAPACK_LIBS) \
   $(BLAS_LIBS) \
   $(READLINE_LIBS) \
   $(TERM_LIBS) \
   $(LIBGLOB) \
   $(REGEX_LIBS) \
   $(DL_LIBS) \
   $(PTHREAD_LIBS) \
+  $(FLIBS) \
   $(LIBS)
 
 LIBOCTAVE_LINK_OPTS = \
   $(SPARSE_XLDFLAGS) \
   $(ARPACK_LDFLAGS) \
   $(QRUPDATE_LDFLAGS) \
   $(FFTW_XLDFLAGS)
-
-if AMCOND_LINK_ALL_DEPS
-  LIBOCTAVE_LINK_DEPS += $(LIBCRUFT_LINK_DEPS)
-  LIBOCTAVE_LINK_OPTS += $(LIBCRUFT_LINK_OPTS)
-endif
diff --git a/liboctave/operators/module.mk b/liboctave/operators/module.mk
--- a/liboctave/operators/module.mk
+++ b/liboctave/operators/module.mk
@@ -41,23 +41,23 @@ OPERATORS_SRC =
 
 OTHER_INC += \
   operators/mx-inlines.cc
 
 OP_SRCDIR = $(srcdir)/operators
 
 ## Special rules for sources which must be built before rest of compilation.
 $(VX_OP_INC) $(VX_OP_SRC) : $(OP_SRCDIR)/mk-ops.awk operators/vx-ops
-	(cd operators; $(AWK) -f ../$(OP_SRCDIR)/mk-ops.awk prefix=vx ../$(OP_SRCDIR)/vx-ops)
+	(cd operators; $(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=vx $(OP_SRCDIR)/vx-ops)
 
 $(MX_OP_INC) $(MX_OP_SRC) : $(OP_SRCDIR)/mk-ops.awk operators/mx-ops
-	(cd operators; $(AWK) -f ../$(OP_SRCDIR)/mk-ops.awk prefix=mx ../$(OP_SRCDIR)/mx-ops)
+	(cd operators; $(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=mx $(OP_SRCDIR)/mx-ops)
 
 $(SMX_OP_INC) $(SMX_OP_SRC) : $(OP_SRCDIR)/sparse-mk-ops.awk operators/sparse-mx-ops
-	(cd operators; $(AWK) -f ../$(OP_SRCDIR)/sparse-mk-ops.awk prefix=smx ../$(OP_SRCDIR)/sparse-mx-ops)
+	(cd operators; $(AWK) -f $(OP_SRCDIR)/sparse-mk-ops.awk prefix=smx $(OP_SRCDIR)/sparse-mx-ops)
 
 operators/mx-ops.h : $(OP_SRCDIR)/mk-ops.awk operators/mx-ops
 	$(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=mx make_inclusive_header=mx-ops.h $(OP_SRCDIR)/mx-ops > $@-t
 	mv $@-t $@
 
 noinst_LTLIBRARIES += operators/liboperators.la
 
 operators_liboperators_la_SOURCES = $(OPERATORS_SRC)
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -17,18 +17,21 @@
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
-  -I$(srcdir)/../libcruft/misc \
   -I$(srcdir)/../liboctave \
+  -I$(srcdir)/../liboctave/array \
+  -I$(srcdir)/../liboctave/cruft/misc \
+  -I$(srcdir)/../liboctave/numeric \
+  -I$(srcdir)/../liboctave/util \
   -I$(srcdir)/../libinterp \
   -I$(top_builddir)/libinterp/interpfcn \
   -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
 OCTAVE_BINARIES = \
   octave \
   octave-cli
 
